src/render/painter.js

```diff
@@ -173,7 +173,11 @@ class Painter {
     }
 
     updateTerrain(style: Style, cameraChanging: boolean) {
-        const enabled = !!style && !!style.terrain && this.transform.projection.name === 'mercator';
+        // FIXME: Move this check as part of the projection interface
```

Comment:
I guess projections could just have an `isGlobeOrMercator` property. It's a bit weird, but probably good enough for now.

Reply:
For that I was thinking to make feature support more explicit, e.g. `projection.supportsTerrain`, similarly to what we have for world copies.